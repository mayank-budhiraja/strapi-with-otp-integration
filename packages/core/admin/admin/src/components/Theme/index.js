import React from 'react';
import { ThemeProvider } from '@strapi/design-system/ThemeProvider';
import PropTypes from 'prop-types';
import { useThemeToggle } from '../../hooks';
import GlobalStyle from '../GlobalStyle';

const Theme = ({ children }) => {
  const { currentTheme, themes } = useThemeToggle();

  return (
    <ThemeProvider theme={themes[currentTheme] || themes.light}>
      {children}
      <GlobalStyle />
    </ThemeProvider>
  );
};

Theme.propTypes = {
  children: PropTypes.element.isRequired,
};

export default Theme;
