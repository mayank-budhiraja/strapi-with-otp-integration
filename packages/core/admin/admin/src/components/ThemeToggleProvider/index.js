/**
 *
 * ThemeToggleProvider
 *
 */

import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { ThemeToggleContext } from '../../contexts';

const THEME_KEY = 'STRAPI_THEME';

const getDefaultTheme = () => {
  const browserTheme = window.matchMedia('(prefers-color-scheme: dark)').matches ? 'dark' : 'light';
  const persistedTheme = localStorage.getItem(THEME_KEY);

  return persistedTheme || browserTheme;
};

const ThemeToggleProvider = ({ children, themes }) => {
  const [currentTheme, setCurrentTheme] = useState(getDefaultTheme());

  const handleChangeTheme = nextTheme => {
    setCurrentTheme(nextTheme);
    localStorage.setItem(THEME_KEY, nextTheme);
  };

  return (
    <ThemeToggleContext.Provider value={{ currentTheme, onChangeTheme: handleChangeTheme, themes }}>
      {children}
    </ThemeToggleContext.Provider>
  );
};

ThemeToggleProvider.propTypes = {
  children: PropTypes.node.isRequired,
  themes: PropTypes.shape({
    light: PropTypes.shape({
      colors: PropTypes.object.isRequired,
      shadows: PropTypes.object.isRequired,
      sizes: PropTypes.object.isRequired,
      zIndices: PropTypes.array.isRequired,
      spaces: PropTypes.array.isRequired,
      borderRadius: PropTypes.string.isRequired,
      mediaQueries: PropTypes.object.isRequired,
      fontSizes: PropTypes.array.isRequired,
      lineHeights: PropTypes.array.isRequired,
      fontWeights: PropTypes.object.isRequired,
    }).isRequired,
    dark: PropTypes.shape({
      colors: PropTypes.object.isRequired,
      shadows: PropTypes.object.isRequired,
      sizes: PropTypes.object.isRequired,
      zIndices: PropTypes.array.isRequired,
      spaces: PropTypes.array.isRequired,
      borderRadius: PropTypes.string.isRequired,
      mediaQueries: PropTypes.object.isRequired,
      fontSizes: PropTypes.array.isRequired,
      lineHeights: PropTypes.array.isRequired,
      fontWeights: PropTypes.object.isRequired,
    }).isRequired,
    custom: PropTypes.object,
  }).isRequired,
};

export default ThemeToggleProvider;
