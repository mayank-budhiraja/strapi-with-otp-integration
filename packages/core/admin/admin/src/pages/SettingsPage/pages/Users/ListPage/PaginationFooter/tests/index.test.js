import React from 'react';
import { render } from '@testing-library/react';
import { IntlProvider } from 'react-intl';
import { createMemoryHistory } from 'history';
import { Router, Route } from 'react-router-dom';
import { lightTheme, darkTheme } from '@strapi/design-system';
import Theme from '../../../../../../../components/Theme';
import ThemeToggleProvider from '../../../../../../../components/ThemeToggleProvider';
import PaginationFooter from '../index';

const makeApp = (history, pagination) => {
  return (
    <IntlProvider messages={{}} textComponent="span" locale="en" defaultLocale="en">
      <ThemeToggleProvider themes={{ light: lightTheme, dark: darkTheme }}>
        <Theme>
          <Router history={history}>
            <Route path="/settings/user">
              <PaginationFooter pagination={pagination} />
            </Route>
          </Router>
        </Theme>
      </ThemeToggleProvider>
    </IntlProvider>
  );
};

describe('DynamicTable', () => {
  it('renders and matches the snapshot', () => {
    const history = createMemoryHistory();
    const pagination = { pageCount: 2 };
    history.push('/settings/user?pageSize=10&page=1&sort=firstname');
    const app = makeApp(history, pagination);

    const { container } = render(app);

    expect(container).toMatchSnapshot();
  });
});
