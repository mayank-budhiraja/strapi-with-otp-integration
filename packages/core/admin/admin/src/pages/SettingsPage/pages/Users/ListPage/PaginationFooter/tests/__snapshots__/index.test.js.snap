// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DynamicTable renders and matches the snapshot 1`] = `
.c29 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c0 {
  padding-top: 16px;
}

.c15 {
  padding-left: 8px;
}

.c1 {
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c17 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c18 > * + * {
  margin-left: 4px;
}

.c21 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c24 {
  font-weight: 600;
  color: #32324d;
  font-size: 0.75rem;
  line-height: 1.33;
}

.c27 {
  color: #32324d;
  font-size: 0.75rem;
  line-height: 1.33;
}

.c25 {
  line-height: revert;
}

.c19 {
  padding: 12px;
  border-radius: 4px;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  outline: none;
}

.c19:after {
  -webkit-transition-property: all;
  transition-property: all;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -4px;
  bottom: -4px;
  left: -4px;
  right: -4px;
  border: 2px solid transparent;
}

.c19:focus-visible {
  outline: none;
}

.c19:focus-visible:after {
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -5px;
  bottom: -5px;
  left: -5px;
  right: -5px;
  border: 2px solid #4945ff;
}

.c22 {
  padding: 12px;
  border-radius: 4px;
  box-shadow: 0px 1px 4px rgba(33,33,52,0.1);
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  outline: none;
}

.c22:after {
  -webkit-transition-property: all;
  transition-property: all;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -4px;
  bottom: -4px;
  left: -4px;
  right: -4px;
  border: 2px solid transparent;
}

.c22:focus-visible {
  outline: none;
}

.c22:focus-visible:after {
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -5px;
  bottom: -5px;
  left: -5px;
  right: -5px;
  border: 2px solid #4945ff;
}

.c23 {
  color: #271fe0;
  background: #ffffff;
}

.c23:hover {
  box-shadow: 0px 1px 4px rgba(33,33,52,0.1);
}

.c26 {
  color: #32324d;
}

.c26:hover {
  box-shadow: 0px 1px 4px rgba(33,33,52,0.1);
}

.c20 {
  font-size: 0.7rem;
  pointer-events: none;
}

.c20 svg path {
  fill: #c0c0cf;
}

.c20:focus svg path,
.c20:hover svg path {
  fill: #c0c0cf;
}

.c28 {
  font-size: 0.7rem;
}

.c28 svg path {
  fill: #666687;
}

.c28:focus svg path,
.c28:hover svg path {
  fill: #4a4a6a;
}

.c7 {
  position: absolute;
  left: 0;
  right: 0;
  bottom: 0;
  top: 0;
  width: 100%;
  background: transparent;
  border: none;
}

.c7:focus {
  outline: none;
}

.c7[aria-disabled='true'] {
  cursor: not-allowed;
}

.c10 {
  padding-right: 16px;
  padding-left: 16px;
}

.c12 {
  padding-left: 12px;
}

.c3 {
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c5 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c8 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c11 {
  color: #32324d;
  display: block;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  font-size: 0.875rem;
  line-height: 1.43;
}

.c4 > * {
  margin-top: 0;
  margin-bottom: 0;
}

.c6 {
  position: relative;
  border: 1px solid #dcdce4;
  padding-right: 12px;
  border-radius: 4px;
  background: #ffffff;
  overflow: hidden;
  min-height: 2rem;
  outline: none;
  box-shadow: 0;
  -webkit-transition-property: border-color,box-shadow,fill;
  transition-property: border-color,box-shadow,fill;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
}

.c6:focus-within {
  border: 1px solid #4945ff;
  box-shadow: #4945ff 0px 0px 0px 2px;
}

.c13 {
  background: transparent;
  border: none;
  position: relative;
  z-index: 1;
}

.c13 svg {
  height: 0.6875rem;
  width: 0.6875rem;
}

.c13 svg path {
  fill: #666687;
}

.c14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background: none;
  border: none;
}

.c14 svg {
  width: 0.375rem;
}

.c9 {
  width: 100%;
}

.c16 {
  color: #666687;
  font-size: 0.875rem;
  line-height: 1.43;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <div>
          <div
            class="c3 c4"
          >
            <div
              class="c5 c6"
            >
              <button
                aria-disabled="false"
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-label="Entries per page"
                aria-labelledby="select-1-label select-1-content"
                class="c7"
                id="select-1"
                type="button"
              />
              <div
                class="c8 c9"
              >
                <div
                  class="c5"
                >
                  <div
                    class="c10"
                  >
                    <span
                      class="c11"
                      id="select-1-content"
                    >
                      10
                    </span>
                  </div>
                </div>
                <div
                  class="c5"
                >
                  <button
                    aria-hidden="true"
                    class="c12 c13 c14"
                    tabindex="-1"
                    type="button"
                  >
                    <svg
                      fill="none"
                      height="1em"
                      viewBox="0 0 14 8"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M14 .889a.86.86 0 01-.26.625L7.615 7.736A.834.834 0 017 8a.834.834 0 01-.615-.264L.26 1.514A.861.861 0 010 .889c0-.24.087-.45.26-.625A.834.834 0 01.875 0h12.25c.237 0 .442.088.615.264a.86.86 0 01.26.625z"
                        fill="#32324D"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="c15"
        >
          <label
            class="c16"
            for="page-size"
          >
            Entries per page
          </label>
        </div>
      </div>
      <nav
        aria-label="pagination"
        class="sc-icMgfS"
      >
        <ul
          class="c17 c18"
        >
          <li>
            <a
              aria-current="page"
              aria-disabled="true"
              class="c19 c20 active"
              href="/settings/user"
              tabindex="-1"
            >
              <div
                class="c21"
              >
                Go to previous page
              </div>
              <svg
                aria-hidden="true"
                fill="none"
                height="1em"
                viewBox="0 0 10 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9.88 14.12L3.773 8 9.88 1.88 8 0 0 8l8 8 1.88-1.88z"
                  fill="#32324D"
                />
              </svg>
            </a>
          </li>
          <li>
            <a
              aria-current="page"
              class="c22 c23 active"
              href="/settings/user?pageSize=10&page=1&sort=firstname"
            >
              <div
                class="c21"
              >
                Go to page 1
              </div>
              <span
                aria-hidden="true"
                class="c24 c25"
              >
                1
              </span>
            </a>
          </li>
          <li>
            <a
              aria-current="page"
              class="c19 c26 active"
              href="/settings/user?pageSize=10&page=2&sort=firstname"
            >
              <div
                class="c21"
              >
                Go to page 2
              </div>
              <span
                aria-hidden="true"
                class="c27 c25"
              >
                2
              </span>
            </a>
          </li>
          <li>
            <a
              aria-current="page"
              aria-disabled="false"
              class="c19 c28 active"
              href="/settings/user?pageSize=10&page=2&sort=firstname"
            >
              <div
                class="c21"
              >
                Go to next page
              </div>
              <svg
                aria-hidden="true"
                fill="none"
                height="1em"
                viewBox="0 0 10 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 1.88L6.107 8 0 14.12 1.88 16l8-8-8-8L0 1.88z"
                  fill="#32324D"
                />
              </svg>
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
  <div
    class="c29"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </div>
</div>
`;
